adapter/cache/refresher.go

```diff
@@ -6,38 +6,112 @@ import (
 	"time"
 )
 
-// DefaultOffset is the default percentage offset of the refresh interval.
-const defaultOffset = 0.05
+// constants defining default option values for the refresher.
+const (
+	defaultFailureThreshold = 2
+	defaultBackoffThreshold = 3
+	defaultOffset           = 0.05
+	defaultInitialBackoff   = 15 * time.Second
+	defaultRepeatedBackoff  = 1 * time.Minute
+	defaultFinalBackoff     = 5 * time.Minute
+)
+
+type Refresh func() (interface{}, error)
 
 // Refresher is a helper service that performs refreshing if the a configured interval has passed.
 type Refresher interface {
 	// Refresh refreshes data if required and returns it.
 	Refresh() (interface{}, error)
 	// Reset cache so next invocation will result in execution of provided refresh function.
 	Reset()
+	// Status returns false if the refreshers failure count exceeded configured threshold and true otherwise.
+	Status() bool
```

Comment:
Returning boolean as status may be ambiguous. Either change it to some constants or rename the method to `IsFailing`